File: q1436/q1436.java
class Solution {
    public String destCity(List<List<String>> paths) {
        Set<String> outgoingCities = new HashSet<>();
        for (List<String> path : paths) {
            outgoingCities.add(path.get(0));
        }

        for (List<String> path : paths) {
            String dest = path.get(1);
            if (!outgoingCities.contains(dest))
                return dest;
        }

        return null;
    }
}

File: q1436/q1436.cpp
class Solution
{
public:
    string destCity(vector<vector<string>> &paths)
    {
        unordered_set<string> outgoingCities;
        for (const vector<string> &path : paths)
        {
            outgoingCities.insert(path[0]);
        }

        for (const vector<string> &path : paths)
        {
            string dest = path[1];
            if (outgoingCities.find(dest) == outgoingCities.end())
                return dest;
        }

        return "";
    }
};